### ISSUE TITLE
Crash when loading certain savestate

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#980]
Attached savestate (rename to oms) causes openMSX to crash when loading.

It seems to be related to the state "FINISH" in the savestate file, which is failing to deserialize:

```
#1  0x000000000041817c in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare (this=0x7fffffffb840, 
    __s=0xf7f38940 <error: Cannot access memory at address 0xf7f38940>)
    at /usr/include/c++/5/bits/basic_string.tcc:1398
        __PRETTY_FUNCTION__ = "int std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::compare(const _CharT*) const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]"
        __size = 6
        __osize = 3
        __len = 3
        __r = -9
#2  0x000000000043de7d in std::operator==<char, std::char_traits<char>, std::allocator<char> > (
    __lhs=0xf7f38940 <error: Cannot access memory at address 0xf7f38940>, 
    __rhs="FINISH") at /usr/include/c++/5/bits/basic_string.h:4925
No locals.
#3  0x0000000000ab1bb3 in openmsx::serialize_as_enum_impl<openmsx::YMF262::EnvelopeState>::fromString (this=0x7fffffffb870, str="FINISH")
    at src/serialize_core.hh:118
        __PRETTY_FUNCTION__ = "T openmsx::serialize_as_enum_impl<T>::fromString(const string&) const [with T = openmsx::YMF262::EnvelopeState; std::__cxx11::string = std::__cxx11::basic_string<char>]"
        p = 0x1405550 <openmsx::mainThreadId>
#4  0x0000000000ab1b1e in openmsx::EnumLoader<openmsx::YMF262::EnvelopeState>::operator()<openmsx::XmlInputArchive, std::tuple<> >(openmsx::XmlInputArchive&, openmsx::YMF262::EnvelopeState&, std::tuple<>, int) (this=0x7fffffffb8d6, 
    ar=..., t=@0x33e5938: openmsx::YMF262::EG_OFF) at src/serialize_core.hh:490
        str = "FINISH"
        sae = {<openmsx::serialize_as_enum_impl<openmsx::YMF262::EnvelopeState>> = {<std::integral_constant<bool, true>> = {static value = true}, 
            info = 0x1405500 <_ZN7openmsxL17envelopeStateInfoE>}, <No data fields>}
#5  0x0000000000ab19d1 in openmsx::InputArchiveBase<openmsx::XmlInputArchive>::serialize<openmsx::YMF262::EnvelopeState> (this=0x7fffffffc530, 
    tag=0xdbfca2 "state", t=@0x33e5938: openmsx::YMF262::EG_OFF)
    at src/serialize.hh:529
        tnc = @0x33e5938: openmsx::YMF262::EG_OFF
        loader = {<if_<openmsx::is_primitive<openmsx::YMF262::EnvelopeState>, openmsx::PrimitiveLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_enum<openmsx::YMF262::EnvelopeState>, openmsx::EnumLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_pointer<openmsx::YMF262::EnvelopeState>, openmsx::PointerLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_collection<openmsx::YMF262::EnvelopeState>, openmsx::CollectionLoader<openmsx::YMF262::EnvelopeState>, openmsx::ClassLoader<openmsx::YMF262::EnvelopeState> > > > >> = {<if_c<false, openmsx::PrimitiveLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_enum<openmsx::YMF262::EnvelopeState>, openmsx::EnumLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_pointer<openmsx::YMF262::EnvelopeState>, openmsx::PointerLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_collection<openmsx::YMF262::EnvelopeState>, openmsx::CollectionLoader<openmsx::YMF262::EnvelopeState>, openmsx::ClassLoader<openmsx::YMF262::EnvelopeState> > > > >> = {<if_<openmsx::serialize_as_enum<openmsx::YMF262::EnvelopeState>, openmsx::EnumLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_pointer<openmsx::YMF262::EnvelopeState>, openmsx::PointerLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_collection<openmsx::YMF262::EnvelopeState>, openmsx::CollectionLoader<openmsx::YMF262::EnvelopeState>, openmsx::ClassLoader<openmsx::YMF262::EnvelopeState> > > >> = {<if_c<true, openmsx::EnumLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_pointer<openmsx::YMF262::EnvelopeState>, openmsx::PointerLoader<openmsx::YMF262::EnvelopeState>, if_<openmsx::serialize_as_collection<openmsx::YMF262::EnvelopeState>, openmsx::CollectionLoader<openmsx::YMF262::EnvelopeState>, openmsx::ClassLoader<openmsx::YMF262::EnvelopeState> > > >> = {<openmsx::EnumLoader<openmsx::YMF262::EnvelopeState>> = {<No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}
```

![machine2](https://cloud.githubusercontent.com/assets/12675978/11104427/fc27e676-88c8-11e5-9b5e-c3e480841fd6.jpg)

**Comment by MBilderbeek** [comment]
Weirdly enough, the EnvelopeState of the YMF262 never has had a "FINISH" state. Was something going wrong when saving state for the YMF262 regarding the enum stuff and was it mixed with some other enum with the same name? (Just a wild guess here.)

**Comment by MBilderbeek** [comment]
The root cause of the wrong state enum value in the savestate remains a mystery, but the crash has been fixed by Wouter in commit 01e471e6917e11df3687e79cd56235fb9f610f59

Should we or should we not close the ticket now?

**Comment by MBilderbeek** [comment]
Let's close it, follow up in the mentioned #985 .
